src/lm/mikolov-rnnlm-lib.h

```diff
@@ -0,0 +1,320 @@
+///////////////////////////////////////////////////////////////////////
+// 
+//  Recurrent neural network based statistical language modeling toolkit
+//  Version 0.3e
+//  (c) 2010-2012 Tomas Mikolov (tmikolov@gmail.com)
+//           2015 Guoguo Chen
+//           2015 Hainan Xu
+// 
+///////////////////////////////////////////////////////////////////////
+
```

Comment:
I have put some text in /home/dpovey/license.h-- please incorporate this
(suitably modified) into the .h and .cc files.



On Thu, Oct 1, 2015 at 3:00 PM, chenguoguo <notifications@github.com> wrote:

> In src/lm/mikolov-rnnlm-lib.h
> <https://github.com/kaldi-asr/kaldi/pull/186#discussion_r40952592>:
>
> > @@ -0,0 +1,320 @@
> > +///////////////////////////////////////////////////////////////////////
> > +//
> > +//  Recurrent neural network based statistical language modeling toolkit
> > +//  Version 0.3e
> > +//  (c) 2010-2012 Tomas Mikolov (tmikolov@gmail.com)
> > +//           2015 Guoguo Chen
> > +//           2015 Hainan Xu
> > +//
> > +///////////////////////////////////////////////////////////////////////
> > +
>
> You are right, here is his full copyright, we should probably copy it to
> the code: Copyright (c) 2010-2012 Tomas Mikolov All rights reserved.
> Redistribution and use in source and binary forms, with or without
> modification, are permitted provided that the following conditions are met:
> 1. Redistributions of source code must retain the above copyright notice,
> this list of conditions and the following disclaimer. 2. Redistributions in
> binary form must reproduce the above copyright notice, this list of
> conditions and the following disclaimer in the documentation and/or other
> materials provided with the distribution. 3. Neither name of copyright
> holders nor the names of its contributors may be used to endorse or promote
> products derived from this software without specific prior written
> permission. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
> CONTRIBUTORS ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
> BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
> FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS OR
> CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
> EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
> PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
> OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
> WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
> OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
> ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> … <#15024d32e5ec89e1_>
> On Thu, Oct 1, 2015 at 2:54 PM, Daniel Povey <notifications@github.com>
> wrote: In src/lm/mikolov-rnnlm-lib.h <
> https://github.com/kaldi-asr/kaldi/pull/186#discussion_r40951851>: > @@
> -0,0 +1,320 @@ >
> +/////////////////////////////////////////////////////////////////////// >
> +// > +// Recurrent neural network based statistical language modeling
> toolkit > +// Version 0.3e > +// (c) 2010-2012 Tomas Mikolov (
> tmikolov@gmail.com) > +// 2015 Guoguo Chen > +// 2015 Hainan Xu > +// >
> +/////////////////////////////////////////////////////////////////////// >
> + Guoguo, we need to have some more clarity on the license here. What did
> the LICENSE file in his distribution say? — Reply to this email directly or
> view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/186/files#r40951851>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/186/files#r40952592>.
>